### DISCUSSION TITLE
[✨] Delta Builds to Add Routes on Live Servers w/o Restart -- Moonshot

### DISCUSSION THREAD

**Discussion by n8sabes** [QwikDev/qwik-evolution#156] [Proposals For Qwik]
### Is your feature request related to a problem?

There is currently no way to add (or remove) routes without rebuilding and redeploying an entire site.

### Describe the solution you'd like

There should be a way to add routes without rebuilding and redeploying the entire site on a live server. New routes would be compiled against a prior build, "adding" new modules to the existing collection.

To update the live server, simply copy the new files. No restart required. This enables new routes that are truly complied to operate against an existing code base on the server, bringing them online live w/o the downtime of a restart. This could be used for automatically generated routes on the fly (via code templates or other pipelines).

`pnpm deltabuild ./prior_build_path`

_It would also be cool if `301` / `302` / `404` could be applied to existing routes in a delta build as well._

### Describe alternatives you've considered

Dynamically generating pages SSR using templates but this defeats the speed and value proposition Qwik offers.

### Additional context

Other possible ways to describe a build process that extends (builds upon, or changes) a prior build:

1. Extend Build
2. Differential Build
3. Delta Build
4. Progressive Build
5. Cumulative Build
6. Evolutive Build

The result would behave like the head of a git repo, or worm storage system.

`pnpm moonshot` 😉

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
